File: binary_search/binary_search.py
def binary_search(array, target):
  left = 0
  right = len(array) - 1
  while left <= right:
    mid = (right + left) // 2
    potential_match = array[mid]
    if target > potential_match:
      left = mid + 1
    elif target < potential_match:
      right = mid - 1
    elif target == potential_match:
      return mid
  return -1

print(binary_search([1, 2, 3, 4, 5, 6], 6))

File: binary_search/binary_search.java
import java.util.*;

public class BinarySearch {
    public static int binarySearch(int[] array, int target) {
        int left = 0;
        int right = array.length - 1;

        while (left <= right) {
            int mid = (right + left) / 2;
            int potentialMatch = array[mid];

            if (target > potentialMatch) {
                left = mid + 1;
            } else if (target < potentialMatch) {
                right = mid - 1;
            } else {
                return mid;
            }
        }

        return -1;
    }

    public static void main(String[] args) {
        int[] numbers = {1, 2, 3, 4, 5, 6};
        int target = 6;
        int result = binarySearch(numbers, target);
        System.out.println(result);
    }
}
